engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -242,19 +254,38 @@ BindableTableScan::BindableTableScan(std::size_t kernel_id, const std::string &
     this->filtered = is_filtered_bindable_scan(expression);
 }
 
-void BindableTableScan::do_process(std::vector< std::unique_ptr<ral::frame::BlazingTable> > inputs,
+ral::execution::task_result BindableTableScan::do_process(std::vector< std::unique_ptr<ral::frame::BlazingTable> > inputs,
     std::shared_ptr<ral::cache::CacheMachine> output,
     cudaStream_t /*stream*/, const std::map<std::string, std::string>& /*args*/) {
     auto & input = inputs[0];
-    if(this->filtered) {
-        auto columns = ral::processor::process_filter(input->toBlazingTableView(), expression, this->context.get());
-        columns->setNames(fix_column_aliases(columns->names(), expression));
+    std::unique_ptr<ral::frame::BlazingTable> columns;
```

Comment:
Can you rename std::unique_ptr<ral::frame::BlazingTable> columns;
to filtered_input, or something more descriptive than just columns. 
Because otherwise we have either input or columns, and from the names its hard to understand what is the difference